src/airflow/dags/common_airflow.py

```diff
@@ -0,0 +1,135 @@
+"""Airflow boilerplate code which can be shared by several DAGs."""
+
+from __future__ import annotations
+
+import os
+
+import pendulum
+from airflow.providers.google.cloud.operators.dataproc import (
+    ClusterGenerator,
+    DataprocCreateClusterOperator,
+    DataprocDeleteClusterOperator,
+    DataprocSubmitJobOperator,
+)
+from airflow.utils.trigger_rule import TriggerRule
+
+# Code version. It has to be repeated here as well as in `pyproject.toml`, because Airflow isn't able to look at files outside of its `dags/` directory.
+otg_version = "1.0.0"
+
+
+# Cloud configuration.
+project_id = "open-targets-genetics-dev"
+region = "europe-west1"
+zone = "europe-west1-d"
+image_version = "2.1"
+google_application_credentials = (
+    "/opt/airflow/config/application_default_credentials.json"
+)
+os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = google_application_credentials
+os.environ["GOOGLE_CLOUD_PROJECT"] = project_id
+
+
+# Executable configuration.
+initialisation_base_path = (
+    f"gs://genetics_etl_python_playground/initialisation/{otg_version}"
+)
+python_cli = f"{initialisation_base_path}/cli.py"
+config_tar = f"{initialisation_base_path}/config.tar.gz"
+package_wheel = f"{initialisation_base_path}/otgenetics-{otg_version}-py3-none-any.whl"
+initialisation_executable_file = [
+    f"{initialisation_base_path}/install_dependencies_on_cluster.sh"
+]
+
+
+# Input/output file configuration.
+version = "XX.XX"
+inputs = "gs://genetics_etl_python_playground/input"
+outputs = f"gs://genetics_etl_python_playground/output/python_etl/parquet/{version}"
+spark_write_mode = "overwrite"
+
+
+# Shared DAG construction parameters.
+shared_dag_args = dict(
+    owner="Open Targets Data Team",
+    retries=3,
+)
+shared_dag_kwargs = dict(
+    tags=["genetics_etl", "experimental"],
+    start_date=pendulum.now(tz="Europe/London").subtract(days=1),
+    schedule_interval="@once",
+    catchup=False,
+)
+
+
+def create_cluster(
+    cluster_name,
+    master_machine_type="n1-standard-4",
+    worker_machine_type="n1-standard-16",
+    num_workers=0,
+):
+    """Generate an Airflow task to create a Dataproc cluster. Common parameters are reused, and varying parameters can be specified as needed."""
+    cluster_generator_config = ClusterGenerator(
+        project_id=project_id,
+        zone=zone,
+        master_machine_type=master_machine_type,
```

Comment:
Actually I think there's an even better way: rather than trying to decide which machine type/cluster size to use for the job, we can configure autoscaling (I've been experimenting with it), and make the cloud worry about it. I think this way is more in line with the Hadoop guidelines.

In any case, would that be all right to leave this change for the next iterations?

Reply:
Sure